connectors/bigtable/flink-connector-gcp-bigtable/src/main/java/com/google/flink/connector/gcp/bigtable/serializers/GenericRecordToRowMutationSerializer.java

```diff
@@ -53,28 +55,49 @@ public class GenericRecordToRowMutationSerializer
         implements BaseRowMutationSerializer<GenericRecord> {
     private final String rowKeyField;
     private final @Nullable String columnFamily;
+    private final Boolean useNestedRowsMode;
 
     /**
      * Constructs a {@code GenericRecordToRowMutationSerializer}.
      *
      * @param rowKeyField The name of the field in the {@link GenericRecord} that represents the
      *     Bigtable <a href="https://cloud.google.com/bigtable/docs/schema-design#row-keys">row
-     *     key</a>.
+     *     key</a>. It must be of type String.
+     * @param useNestedRowsMode Whether to use nested rows mode. If {@code true}, each field in the
+     *     {@link GenericRecord} (except the row key field) represents a separate column family.
+     *     Otherwise, all fields are written to a single column family specified by {@code
+     *     columnFamily}.
      * @param columnFamily The name of the <a
      *     href="https://cloud.google.com/bigtable/docs/schema-design#column-families">column
      *     family</a> to use (only in column family mode).
      */
     public GenericRecordToRowMutationSerializer(
-            String rowKeyField, Boolean useNestedRows, @Nullable String columnFamily) {
+            String rowKeyField, Boolean useNestedRowsMode, @Nullable String columnFamily) {
         this.rowKeyField = rowKeyField;
         this.columnFamily = columnFamily;
+        this.useNestedRowsMode = useNestedRowsMode;
     }
 
     @Override
     public RowMutationEntry serialize(GenericRecord record, SinkWriter.Context context) {
         RowMutationEntry entry = RowMutationEntry.create((String) record.get(rowKeyField));
 
-        return serializeWithColumnFamily(record, entry, this.columnFamily, context);
+        if (!useNestedRowsMode) {
+            return serializeWithColumnFamily(record, entry, this.columnFamily, context);
+        }
+
+        for (Schema.Field field : record.getSchema().getFields()) {
+            String columnFamily = field.name();
+            if (!columnFamily.equals(rowKeyField)) {
```

Comment:
in the else, let's log that we're skipping this field because it's the rowkey field

Reply:
This log per element, I am not sure if it's a good idea.